src/hotspot/share/oops/oop.hpp

```diff
@@ -311,4 +320,11 @@ class oopDesc {
   DEBUG_ONLY(bool get_UseG1GC();)
 };
 
+// An oopDesc is not initialized via a constructor.  Space is allocated in
+// the Java heap, and static functions provided here on HeapWord* are used
+// to fill in certain parts of that memory.  For that to be valid, the
+// object must not have non-trivial initialization (C++14 3.8).  For that to
```

Comment:
Can we avoid the double-negative and state it "must have trivial initialization"?

Reply:
I've improved the wording, including eliminating the double negative.  I followed the wording in the standard a little too closely; that's where "non-trivial initialization" came from. I think "must have trivial initialization" isn't right, since "initialization" in this context involves invocation of a constructor, and we're dealing with a case where that doesn't happen.